src/aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -269,80 +335,19 @@ def inspect_relax(self):
-        self.report(f'after iteration {self.ctx.iteration} cell volume of relaxed structure is {curr_cell_volume}')
-
-        # After first iteration, simply set the cell volume and restart the next base workchain
-        if not prev_cell_volume:
-            self.ctx.current_cell_volume = curr_cell_volume
-
-            # If meta convergence is switched off we are done
-            if not self.ctx.meta_convergence:
-                self.ctx.is_converged = True
-            return
-
-        # Check whether the cell volume is converged
-        volume_threshold = self.inputs.volume_convergence.value
-        volume_difference = abs(prev_cell_volume - curr_cell_volume) / prev_cell_volume
-
-        if volume_difference < volume_threshold:
-            self.ctx.is_converged = True
-            self.report(
-                f'relative cell volume difference {volume_difference} smaller than threshold {volume_threshold}'
-            )
-        else:
-            self.report(
-                f'current relative cell volume difference {volume_difference} larger than threshold {volume_threshold}'
-            )
-
-        self.ctx.current_cell_volume = curr_cell_volume
-
-        return
-
-    def run_final_scf(self):
-        """Run the `PwBaseWorkChain` to run a final scf `PwCalculation` for the relaxed structure."""
-        inputs = self.ctx.final_scf_inputs
-        inputs.pw.structure = self.ctx.current_structure
-
-        inputs_nbnd = inputs.pw.parameters.get('SYSTEM', {}).get('nbnd', None)
-        if self.ctx.current_number_of_bands is not None and inputs_nbnd is None:
-            inputs.pw.parameters.setdefault('SYSTEM', {})['nbnd'] = self.ctx.current_number_of_bands
-
-        inputs = prepare_process_inputs(PwBaseWorkChain, inputs)
-        running = self.submit(PwBaseWorkChain, **inputs)
-
-        self.report(f'launching PwBaseWorkChain<{running.pk}> for final scf')
-
-        return ToContext(workchain_scf=running)
-
-    def inspect_final_scf(self):
-        """Inspect the result of the final scf `PwBaseWorkChain`."""
-        workchain = self.ctx.workchain_scf
-
-        if not workchain.is_finished_ok:
-            self.report(f'final scf PwBaseWorkChain failed with exit status {workchain.exit_status}')
-            return self.exit_codes.ERROR_SUB_PROCESS_FAILED_FINAL_SCF
 
     def results(self):
         """Attach the output parameters and structure of the last workchain to the outputs."""
-        if self.ctx.is_converged and self.ctx.iteration <= self.inputs.max_meta_convergence_iterations.value:
-            self.report(f'workchain completed after {self.ctx.iteration} iterations')
-        else:
-            self.report('maximum number of meta convergence iterations exceeded')
-
         # Get the latest relax workchain and pass the outputs
-        final_relax_workchain = self.ctx.workchains[-1]
+        final_relax_workchain = self.ctx.base_relax_workchains[-1]
```

Comment:
You can remove the try-except on line 355. The `self.ctx.workchain_scf` is never set as that used to be done in the final scf.